### PR TITLE
Add codeowners file

### PR DESCRIPTION
**Pull request by flomonster**
Here's a suggestion for adding code owners to files so each project can manage its org autonomously.

To make this PR useful we should change a few repository settings:

- Add branch protection for `main`:
  - Require a pull request before merging
  - Require review from Code Owners
  - Require status check to pass before merging
  - Require conversation resolution before merging 
  - Require linear history 
- Add write permissions to openrailassociation members

> [!NOTE]
> We could also add a check to ensure that the yaml file doesn't affect any groups other than the one for which it was created (`OSRD` group for the `osrd.yaml` file and so on...).

### CODE REVIEW

**Comment by mxmehl** [comment]
Thanks for the suggestion!

My experience with CODEOWNERS is limited. I understand that the OSRD-Admin team would be able merge PRs for the given file without any additional approval. Could this team also merge other files than the `osrd.yaml`?

**Comment by flomonster** [comment]
No, we won't be able to merge other files without your approval.
The suggested `CODEOWNERS` file can be read as follows:

- All file changes require @mxmehl approval
- Except `osrd.yml`, which requires approval from @OpenRailAssociation/osrd-admins 

Suppose someone makes changes to both `osrd.yml` and another file. The PR will require approval from both you and someone from osrd-admin.

**Comment by cornelius** [comment]
I think this is a good way to manage team memberships. We do want the projects to be able to manage their team members without requiring a central approval all the time. So I would be fine with trying it with the CODEOWNERS mechanism and see how it goes.

**Comment by mxmehl** [comment]
This topic has been [discussed in the TC yesterday](https://github.com/OpenRailAssociation/technical-committee/pull/75) to make sure this is a common understanding and can be applied to all OpenRail projects. The feedback, as Cornelius wrote, was positive. I will implement and document it soon.

**Review by mxmehl — APPROVED** [review]
LGTM, as discussed in TC. I just added Cornelius (TC Chair) as additional global Codeowner

**Comment by mxmehl** [comment]
Damn, I just saw the requirement "Add write permissions to openrailassociation members" isn't trivial as it seems. I don't want to change base permissions of all organization members to all repos to "write".

@flomonster, do you know a way to enable all org members to write to a single repo? I just found a workaround of adding all org members to a team which is really hacky.

**Comment by flomonster** [comment]
@mxmehl, I don't know a way to do it. I see another workaround, you could add write permission for the OSRD team (you will have to add each new group). It's cleaner but requires a bit of maintenance.

**Comment by mxmehl** [comment]
> @mxmehl, I don't know a way to do it. I see another workaround, you could add write permission for the OSRD team (you will have to add each new group). It's cleaner but requires a bit of maintenance.

That's a good idea, thanks!

**Pull request merged**
